protocol/packages/swap/src/utils.rs

```diff
@@ -0,0 +1,59 @@
+use currency::Group;
+use finance::coin::{CoinDTO, NonZeroAmount};
+#[cfg(feature = "astroport")]
+use sdk::cosmos_sdk_proto::{cosmos::base::v1beta1::Coin as ProtoCoin, prost::Name};
+use sdk::cosmos_sdk_proto::{prost::Message, Any};
+
+#[cfg(feature = "astroport")]
+pub(crate) fn parse_request_from_any<T>(request: Any) -> T
+where
+    T: Message + Default + Name,
+{
+    request.to_msg().expect("Expected a swap request message!")
+}
+
+#[cfg(feature = "osmosis")]
+pub(crate) fn parse_request_from_any_and_type_url<T>(request: Any, type_url: &str) -> T
+where
+    T: Message + Default,
+{
+    assert_eq!(
+        request.type_url, type_url,
+        "Different type URL than expected one encountered!"
+    );
+
+    T::decode(request.value.as_slice()).expect("Expected a swap request message!")
+}
+
+pub(crate) fn parse_token<G>(amount: &str, denom: String) -> CoinDTO<G>
```

Comment:
since it is only one fn we may not be able to introduce a module with a name to classify the functionality defined in, then move it up to the lib.rs